support-lib/CMakeLists.txt

```diff
@@ -0,0 +1,76 @@
+cmake_minimum_required(VERSION 3.6.0)
+
+project(Djinni)
+
+include(GNUInstallDirs)
+
+set(SRC_SHARED
+  "djinni_common.hpp"
+  "proxy_cache_interface.hpp"
+  "proxy_cache_impl.hpp"
+)
+
+set(SRC_JNI
+  "jni/djinni_support.hpp"
+  "jni/Marshal.hpp"
+  "jni/djinni_support.cpp"
+)
+
+set(SRC_OBJC
+  "objc/DJICppWrapperCache+Private.h"
+  "objc/DJIError.h"
+  "objc/DJIMarshal+Private.h"
+  "objc/DJIObjcWrapperCache+Private.h"
+  "objc/DJIError.mm"
+  "objc/DJIProxyCaches.mm"
+)
+
+option(DJINNI_STATIC_LIB "Build Djinni as a static library instead of dynamic (the default)." off)
+if(DJINNI_STATIC_LIB)
+  add_library(djinni STATIC ${SRC_SHARED})
```

Comment:
Ah, the case of a single library with both languages is one I hadn't thought about, since it doesn't come up in our use cases.
I wonder about the converse case with two different libraries by the same name, used on different platforms.  I guess in most build systems the platforms might be separated enough on their own to not cause confusion?  In our use cases, we use distinct names to avoid confusion in shared files, but since we use different build tools on most platforms, the only real confusion occurs between iOS and Mac, which both use Xcode.